services/agents/registry.go

```diff
@@ -187,7 +194,18 @@ func (r *Registry) Run(stream agentpb.Agent_ConnectServer) error {
 				})
 
 			case *agentpb.ActionResultRequest:
-				panic("TODO")
+				agent.channel.SendResponse(&channel.ServerResponse{
+					ID:      req.ID,
+					Payload: new(agentpb.ActionResultResponse),
+				})
+				// TODO: PMM-3978: In the future we need to merge action parts before send it to storage.
+				r.actionsStorage.Store(models.ActionResult{
+					ID:         p.ActionId,
+					PmmAgentID: agent.id,
+					Done:       p.Done,
+					Error:      p.Error,
+					Output:     string(p.Output),
+				})
```

Comment:
Note that other cases do stuff first, then send a response. Please do the same.

It also makes sense to check if `Done` is `true`, and log a warning otherwise. That will make it more noticeable for us later.